src/Magnum/Math/Bezier.h

```diff
@@ -0,0 +1,118 @@
+                _points = points;
+            }
+
+            /**
+             * @brief Divides a Bezier curve into two curves of same order having their own control points.
+             *        De Casteljau's algorithm is used.
+             * @param t The interpolation factor
+             *
+             * @return Array of two Bezier curves of the same order
+             */
+            std::array<Bezier<order, dimensions, T >, 2> subdivide(Float t) const {
+                auto i_points = calculateIntermediatePoints(t);
+                std::array<Vector<dimensions, T>, order + 1> left, right;
+                for(UnsignedInt i=0; i<=order; ++i){
+                    left[i] = i_points[0][i];
+                }
+                for(UnsignedInt i = 0, j = order; j>=0; --j, ++i){
+                    right[i] = i_points[i][j];
+                }
+                return {Bezier<order, dimensions, T>(left), Bezier<order, dimensions, T>(right)};
+            }
+
+            /**
+             * @brief Finds the point in the curve for a given interpolation factor
+             *        De Casteljau's algorithm is used.
+             * @param t The interpolation factor
+             */
+            Vector<dimensions, T> lerp (Float t) const {
+                auto i_points= calculateIntermediatePoints(t);
+                return i_points[0][order];
+            };
+
+        private:
+
+            /**
+             * @brief Calculates and returns all intermediate points generated when using De Casteljau's algorithm
+             * @param t The interpolation factor
+             *
+             */
+            std::array<std::array<Vector<dimensions,T>, order + 1>, order + 1> calculateIntermediatePoints(Float t) const {
+                const auto n = order + 1;
+                std::array<std::array<Vector<dimensions,T>, n>, n> i_points;
+                for (UnsignedInt i = 0; i < n; ++i) {
+                    i_points[i][0] = _points[i];
+                }
+                for (UnsignedInt r = 1; r < n; ++r) {
+                    for (UnsignedInt i = 0; i < n - r; ++i) {
+                        i_points[i][r] = (1 - t) * i_points[i][r - 1] + t * i_points[i + 1][r - 1];
+                    }
+                }
+                return i_points;
+            };
+
+            std::array<Vector<dimensions, T >, order + 1> _points;
+        };
+
+        template<UnsignedInt dimensions, class T> using QuadraticBezier = Bezier<2, dimensions, T>;
+        template<UnsignedInt dimensions, class T> using CubicBezier = Bezier <3, dimensions, T>;
+        template<class T> using QuadraticBezier2D = QuadraticBezier<2, T>;
+        template<class T> using QuadraticBezier3D = QuadraticBezier<3, T>;
+        template<class T> using CubicBezier2D = CubicBezier<2, T>;
+        template<class T> using CubicBezier3D = CubicBezier<3, T>;
+
+    }
+
+}
+#endif //Magnum_Math_Bezier_h
```

Comment:
Magnum code usually doesn't annotated `#endif`s, since these kinds of comments are likely to be missed during refactoring (and it's rather obvious what this endif does anyway)